### PR TITLE
[bugfix] preserve absent KJT weights in exported graphs

### PR DESCRIPTION
**Pull request by eric-gecheng**
FBGEMM CUDA can return an undefined tensor for absent weights, causing native TorchScript inference to fail on a subsequent permutation. Insert FX guards after TorchRec MC and quantized permutations so unweighted inputs retain None in exported models.

### CODE REVIEW

File: tzrec/utils/fx_util.py [hunk 1]
```diff
@@ -13,15 +13,32 @@
...
 # turn them into Python ints -- so tracing keeps them opaque and TorchScript
 # compiles them whole. Matched by class name.
 UNTRACEABLE_MODULES = ["ComputeJTDictToKJT", "PromptAssembler", "HoleKeyBuilder"]
 
 
+@torch.fx.wrap
+def _restore_unweighted_kjt(
+    source: KeyedJaggedTensor, permuted: KeyedJaggedTensor
+) -> KeyedJaggedTensor:
+    """Preserve absent weights after a TorchRec feature permutation.
```
[the remark this answers is not in the document]
Comment (eric-gecheng) [reply tzrec/utils/fx_util.py:29]: Addressed in [701fbc5](https://github.com/alibaba/TorchEasyRec/pull/679/commits/701fbc54a6c76c904c7d3958d4707a96e0756c4f). The docstring now explicitly states that the helper mutates and returns the same `permuted` object, and that `source` may alias it on identity-permutation paths. I also checked same-object calls with both absent and present weights; object identity and valid weights are preserved.

File: tzrec/utils/fx_util.py [hunk 2]
```diff
@@ -48,7 +65,28 @@ def symbolic_trace(
     _leaf_modules = list(UNTRACEABLE_MODULES)
     if leaf_modules:
         _leaf_modules.extend(leaf_modules)
-    return _symbolic_trace(root, concrete_args, _leaf_modules)
+    gm = _symbolic_trace(root, concrete_args, _leaf_modules)
```
[the remark this answers is not in the document]
Comment (eric-gecheng) [reply tzrec/utils/fx_util.py:76]: Addressed in [701fbc5](https://github.com/alibaba/TorchEasyRec/pull/679/commits/701fbc54a6c76c904c7d3958d4707a96e0756c4f). The `symbolic_trace` docstring now describes insertion of absent-weight guards after FX-wrapped TorchRec KJT permutations and the idempotent behavior when tracing an already guarded graph.

File: tzrec/utils/fx_util.py [hunk 3]
```diff
@@ -48,7 +65,28 @@ def symbolic_trace(
...
+            user = next(iter(node.users))
+            if user.target == _restore_unweighted_kjt and user.args == (source, node):
+                continue
+        with gm.graph.inserting_after(node):
+            restored = gm.graph.call_function(
+                _restore_unweighted_kjt, args=(source, node)
+            )
+        restored.meta["is_wrapped"] = True
+        node.replace_all_uses_with(restored)
+        restored.args = (source, node)
```
[the remark this answers is not in the document]
Comment (eric-gecheng) [reply tzrec/utils/fx_util.py:98]: Addressed in [701fbc5](https://github.com/alibaba/TorchEasyRec/pull/679/commits/701fbc54a6c76c904c7d3958d4707a96e0756c4f). Added short comments explaining the re-trace deduplication, why `is_wrapped` must be retained, and why the guard input must be restored after `replace_all_uses_with` to avoid a self-loop.

An `inserted` flag now skips the extra `lint()` and `recompile()` when the graph was unchanged. A focused check confirmed that the post-processing calls neither on a graph without these permutations, and the existing re-trace regression still passes.

**Comment by eric-gecheng** [comment]
Follow-up to the [review summary](https://github.com/alibaba/TorchEasyRec/pull/679#issuecomment-5748069418): changes are pushed in [701fbc5](https://github.com/alibaba/TorchEasyRec/pull/679/commits/701fbc54a6c76c904c7d3958d4707a96e0756c4f), with replies on all three inline comments.

1. **Private TorchRec APIs:** moved the two private helper imports inside `symbolic_trace`, limiting their import-time dependency to tracing. The pinned TorchRec version and the necessary `KJT._weights` normalization remain unchanged.

2. **Quantization and unified AOTI:** added permanent `torch.export` regression cases with and without weights. Separately tested a CUDA INT8 MC-EBC model with shared tables and mixed SUM/MEAN pooling under `QUANT_EMB=1 ENABLE_AOT=2`: FX tracing, `torch.export`, and exported-model execution passed, and the guarded/unguarded ATen graphs were identical. The existing unified-AOT integration test does not explicitly disable EBC quantization; these additional checks target the guarded permutation path. Full AOTI binary compilation/package loading remains unverified because the matching local Torch 2.13 environment lacks the CUDA 13 development toolkit.

3. **Test-side `torch.fx.wrap` registrations:** retained them and documented why. Wrap registration is tied to the caller's globals; definition-site wrapping in TorchRec does not register these imported aliases in the test module. In an isolated probe, removing the two registrations changed the graph to two `call_method("permute")` nodes and produced zero guards, so they are not redundant.

4. **Timeout diagnostics:** `TimeoutExpired` already retains captured output, but the default test traceback did not display it. The test now reports decoded stdout/stderr and the child flushes stage/case progress. An injected timeout confirmed that both streams are included.

Validation: all 10 tests in `tzrec.utils.fx_util_test` passed; full `pre-commit run -a` passed; project-wide `pyrefly check` reported 0 errors.

**Review by tiankongdeguiji — APPROVED** [review]
